src/HttpHeaderTools.cc

```diff
@@ -233,6 +234,19 @@ httpHeaderParseQuotedString(const char *start, const int len, String *val)
     return 1;
 }
 
+/**
+ * Temporary bridge for the String-based variant to support conversion
+ * from SquidString to SBuf
+ * TODO: make this the primary implementation
+ */
+int httpHeaderParseQuotedString(const char *start, const int len, SBuf &val)
+{
+    String s;
+    auto rv = httpHeaderParseQuotedString(start, len, &s);
+    val = StringToSBuf(s);
```

Comment:
Doing what you suggest would balloon the scope of the PR.

I would prefer an approach where
1. data members and simple interfaces get ported. There are 58 files including SquidString right now
2. once they are ported, get rid of the bridge methods like this one; one PR per method, building unit tests first to ensure non regressions

Reply:
> Doing what you suggest would balloon the scope of the PR.

The above assertion is false because suggested option 1 keeps the scope of this PR exactly the same.

> I would prefer an approach where [we first introduce bridge functions and then get rid of them].

I am not against bridge functions per se, and suggested option 1 arguably continues to use a bridge function. A bridge function is the best way forward in many cases, but each suggested conversion should be evaluated on its own. Attempts to confine all future conversions to one conversion technique (e.g., bridge functions) seem counter productive to me because Squid code is too varied for one technique to be the right choice in all conversion cases.

As you know, I support minimal conversion PRs and even prefer those whenever they are feasible. Needless to say, there are many ways to convert and even many ways to create bridge functions. Naturally, there are certain principles or invariants we should observe _regardless_ of the PR size or the overall conversion approach. This change request is about violations of some of those principles: 

* avoid performance regressions (option 2 avoids such a regression but requires more work)
* if you have to create a performance regression, let the potential caller know (option 1 does that)

Reply:
> > Doing what you suggest would balloon the scope of the PR.
> 
> The above assertion is false because suggested option 1 keeps the scope of this PR exactly the same.

You are right; I had improperly focused only on part of your answer.

> > I would prefer an approach where [we first introduce bridge functions and then get rid of them].
> 
> I am not against bridge functions per se, and suggested option 1 arguably continues to use a bridge function. A bridge function is the best way forward in many cases, but each suggested conversion should be evaluated on its own. Attempts to confine all future conversions to one conversion technique (e.g., bridge functions) seem counter productive to me because Squid code is too varied for one technique to be the right choice in all conversion cases.
> 
> As you know, I support minimal conversion PRs and even prefer those whenever they are feasible. Needless to say, there are many ways to convert and even many ways to create bridge functions. Naturally, there are certain principles or invariants we should observe _regardless_ of the PR size or the overall conversion approach. This change request is about violations of some of those principles:
> 
> * avoid performance regressions (option 2 avoids such a regression but requires more work)
> * if you have to create a performance regression, let the potential caller know (option 1 does that)

I'll for for option 1 for now. I would like to push through getting rid of SquidStrig

Reply:
Some of these are borne out of wanting to avoid issues such as [this other comment](https://github.com/squid-cache/squid/pull/1451#discussion_r1289147284), coupled with the opportunity that these methods have few callers. I'll see where it leads and that it doesn't blow the scope up to much

Reply:
> I'll see where it leads and that it doesn't blow the scope up to much

Sounds like a good plan to me. When assessing where these changes have led, please keep in mind the essence of my previous [comment](https://github.com/squid-cache/squid/pull/1451#discussion_r1291613871): The added long-term API changes will need to be adjusted before they can land -- there is currently a significant amount of expected additional work in this PR. triggered by those long-term API changes.